transcom/mymove
# B-18393-INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18393)

## Summary

PPMs for moves whose origin duty location does not provide service counseling are counseled by PRIME. Because PRIME does not have the authority to approve or reject AOA requests, customers will have to do this manually. As such, when a customer requests an advance while setting up their PPM, our system will have that flagged as "needing review" when it eventually makes its way to a closeout counselor. This is confusing for them.

As such it was determined that if a prime counseled customer requests an advance, and indicates it was received, they closeout counselor should see "Received" in their UI.
![image](https://github.com/transcom/mymove/assets/147535220/7ca36564-1e9c-44fd-ae2f-bf8c6991d966)
If a prime counseled customer requests an advance, and indicates it was not received, the closeout counselor should see "Not received" in their UI.
![image](https://github.com/transcom/mymove/assets/147535220/46570d54-eac8-4940-aa6b-7fb759ef5088)

### How to test

1. Create a new move that will be prime counseled (for example, use NS Mayport 32228 as your current duty location)
2. Create a PPM and request an advance
3. Login as a TOO and approve the move for counseling
4. Login as PRIME and indicate counseling is complete
5. Login as the customer and upload your PPM documents. When you are asked if you received an advance, indicate no and save. On the next screen, enter your trip details (vehicle, weights, etc) and save
6. Login as a QAE and view the move. The advance request status should say "Not received" (like the 2nd screenshot above)
7. Login as the customer and continue your PPM upload. Click the "Edit" text in the "About your PPM" section.
![image](https://github.com/transcom/mymove/assets/147535220/7b6eea48-4233-416d-a67f-907d4292fcaf)
8. Indicate that you did receive an advance and save
9. Login as a QAE and view the move. The advance request status should say "Received"
